independent-projects/arc/runtime/src/main/java/io/quarkus/arc/AbstractAnnotationLiteral.java

```diff
@@ -0,0 +1,15 @@
+package io.quarkus.arc;
+
+import java.lang.annotation.Annotation;
+
+/**
+ * Redeclares some {@link Annotation} methods to avoid interface type checks / casts.
```

Comment:
Could you be more specific? How does it help?

Reply:
It helps because the check is made against an abstract class instead of an interface. See https://github.com/RedHatPerf/type-pollution-agent for more details

Reply:
I am sure we can get @franz1981 to do an extensive essay on the reason if needed :)

Reply:
Let's call it "insights" eheh

Reply:
I suggest to share the reports you got or the original issue on the pr to benefits other

Reply:
Essentially this PR is the foundation for a following change that deals with Cache annotation instances of the type pollution reported [here](https://github.com/quarkusio/quarkus/issues/28795#issuecomment-1293184173).

Reply:
Ok, could we add some more info to the javadoc? Ideally, a little explanation + link to https://github.com/RedHatPerf/type-pollution-agent.

Reply:
Sounds like a good idea to me

Reply:
I'll try to write something down.

Reply:
Added an essay on this :-)